src/derivative_operators/robin_bc_extended.jl

```diff
@@ -0,0 +1,78 @@
+
+# For a boundary condition, the variables correspond to al*u(0) + bl*u'(0) = cl
+struct RobinBC{T}
+    al::T
+    bl::T
+    cl::T
+    dx_l::T # should grid size be Real or more general?
+    ar::T
+    br::T
+    cr::T
+    dx_r::T
+
+    function RobinBC(al::T, bl::T, cl::T, dx_l::T, ar::T, br::T, cr::T, dx_r::T) where T
+        return new{T}(al, bl, cl, dx_l, ar, br, cr, dx_r)
+    end
+end
+
+struct RobinBCExtended{T,T2<:AbstractVector{T}}
+    l::T
+    r::T
+    u::T2
+
+    function RobinBCExtended(u::T2, al::T, bl::T, cl::T,
+                                    dx_l::T, ar::T, br::T, cr::T, dx_r::T) where
+                                    {T,T2<:AbstractVector{T}}
+        u                    = u
+        l = (cl - bl*u[1]/dx_l)*(1/(al-bl/dx_l))
```

Comment:
Notice how this is just encoding `u' = (u[1] - u[0]) / dx`. The next step will be to generalize this to higher stencils.